JS/script.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable max-classes-per-file */
+/* eslint-disable class-methods-use-this */
+let bookTitle;
+let bookAuthor;
+const library = document.querySelector('.library');
+const bookList = JSON.parse(localStorage.getItem('bookList')) || [];
+
+function getInformation() {
+  bookTitle = document.getElementById('bookTitle').value;
+  bookAuthor = document.getElementById('bookAuthor').value;
+}
+
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+}
+
+class BookShelf {
+  constructor() {
+    this.head = null;
+    this.tail = null;
+    this.length = 0;
+  }
+
+  addBook(title, author) {
+    const newBook = new Book(title, author);
+    bookList.push(newBook);
+    localStorage.setItem('bookList', JSON.stringify(bookList));
+    this.get();
+  }
+
+  deleteBook(title) {
+    const resultBooks = bookList.filter((book) => book.title !== title);
+    localStorage.setItem('bookList', JSON.stringify(resultBooks));
+    window.location.reload();
+  }
+
+  get() {
+    let bookgenerator = '';
+    bookList.forEach((book) => {
+      bookgenerator += `<div class="book-container">
+             <p class="book-title">"${book.title}" by ${book.author}</p>
+             <button class="remove-book" id="${book.title}" onClick=giveInformation(this)>Remove</button>
+           </div>`;
+    });
+    library.innerHTML = bookgenerator;
+  }
+}
+
+const shelf = new BookShelf();
+const addButton = document.querySelector('.add-button');
+const deleteButton = document.querySelector('.remove-book');
+addButton.addEventListener('click', () => {
+  getInformation();
+  shelf.addBook(bookTitle, bookAuthor);
+});
+function giveInformation(id) {
+  const index = id.id;
+  shelf.deleteBook(index);
+}
+window.onload = () => {
+  let bookgenerator = '';
+  bookList.forEach((book) => {
+    bookgenerator += `<div class="book-container">
+                 <p class="book-title">"${book.title}" by ${book.author}</p>
+                 <button class="remove-book" id="${book.title}" onClick=giveInformation(this)>Remove</button>
+              </div>`;
+  });
+  library.innerHTML = bookgenerator;
+};
+
+const x = 0;
+if (x === 1) {
+  giveInformation();
+  deleteButton();
```

Comment:
[OPTIONAL] I think this block of code is not really necessary. Please consider removing it if you are not going to use this. :eyes: :eyes:

Reply:
Hi @Santiago220991, it is unnecessary indeed, we added just to avoid the linter error of 'function defined but not used' since the button is inside the html generator div,  would you suggest adding a 'ignore the rule' command for the linters instead?

Reply:
Hi, @Gdiazdiaz I think you are also not using the giveInformation and deleteButton functions. If you remove the functions you will not need to add that block of code.

Reply:
@Santiago220991 We are thou, if you check the code we call the givenInformation function inside the generated html and it's a little filter than then passes the information to delete the book, even if we didn't have that filter it will be the deleteBook function the one that will not be called and the linter error will still be present, the only two ways I can think of to fix it is the useless block or the ignore command, wich one do you think is better or if you have a third option please let us know.

Reply:
I think you are using the getInformation function the givenInformation function is just declared. However, this is just an optional requirement so you can omit it.

Reply:
That's because, getInformation gets called in a normal js line and the givenInformation gets called in the generated html div that's why the linter says declared but not used, it is being used but because it's getting called inside an html generator is not getting recognized, I know it's optional but I was expecting your insight because I had this issue in my other reps with js generated divs and I would appreciate to have your opinion on wich solution is better or a third option if you can think on an alternative.

Reply:
@Gdiazdiaz You're right I didn't notice that. You could try adding an event listener instead of adding an onClick property using HTML.